pkg/kubectl/resource/visitor.go

```diff
@@ -463,9 +452,20 @@ func NewStreamVisitor(r io.Reader, mapper *Mapper, source string, schema validat
 	}
 }
 
+func (v *StreamVisitor) SetStreamTransform(fn StreamTransform) {
+	v.Transform = fn
+}
+
 // Visit implements Visitor over a stream. StreamVisitor is able to distinct multiple resources in one stream.
 func (v *StreamVisitor) Visit(fn VisitorFunc) error {
-	d := yaml.NewYAMLOrJSONDecoder(v.Reader, 4096)
+	reader := v.Reader
```

Comment:
Why does the transformer have to be inline?  Don't you want to transform *after* we normalize to JSON?